lib/charms/parca_k8s/v0/parca_scrape.py

```diff
@@ -382,53 +359,13 @@ def __init__(self, charm: ops.CharmBase, relation_name: str = DEFAULT_RELATION_N
             RelationInterfaceMismatchError: The relation with the same name as provided
                 via `relation_name` argument does not have the `parca_scrape` relation
                 interface.
-            ops.RelationRoleMismatchError: If the relation with the same name as provided
+            RelationRoleMismatchError: If the relation with the same name as provided
                 via `relation_name` argument does not have the `ops.RelationRole.requires`
                 role.
         """
-        _validate_relation_by_interface_and_direction(
-            charm, relation_name, RELATION_INTERFACE_NAME, ops.RelationRole.requires
-        )
-
-        super().__init__(charm, relation_name)
-        self._charm = charm
         self._relation_name = relation_name
-        events = self._charm.on[relation_name]
-        self.framework.observe(
-            events.relation_changed, self.on_profiling_provider_relation_changed
-        )
-        self.framework.observe(
-            events.relation_departed, self._on_profiling_provider_relation_departed
-        )
-
-    def on_profiling_provider_relation_changed(self, event):
-        """Handle changes with related profiling providers.
-
-        Anytime there are changes in relations between Parca and profiling provider charms the
-        Parca charm is informed, through a `TargetsChangedEvent` event. The Parca charm can then
-        choose to update its scrape configuration.
 
-        Args:
-            event: a `CharmEvent` resulting in the Parca charm updating its scrape configuration
-        """
-        rel_id = event.relation.id
-
-        self.on.targets_changed.emit(relation_id=rel_id)
-
-    def _on_profiling_provider_relation_departed(self, event):
-        """Update job config when a profiling provider departs.
-
-        When a profiling provider departs the Parca charm is informed through a
-        `TargetsChangedEvent` event so that it can update its scrape configuration to ensure that
-        the departed profiling provider is removed from the list of scrape jobs.
-
-        Args:
-            event: a `CharmEvent` that indicates a profiling provider unit has departed.
-        """
-        rel_id = event.relation.id
-        self.on.targets_changed.emit(relation_id=rel_id)
-
-    def jobs(self) -> list:
+    def jobs(self, charm: ops.CharmBase) -> list:
```

Comment:
I like the concept of not providing the charm to the library, but in this case we've just moved the `charm` from `__init__` to `jobs()` (the only public method).  This seems like a change without a real impact other than adding the chore of another arg to `jobs`.  This would be more compelling if we handled multiple charms with one lib maybe, but that isn't a concern.

If the goal is to avoid passing the charm, how about passing `charm.model.relations` instead?  That decreases the potential for coupling, but leaves the bit that actually matters.  My preference would be to pass `charm.model.relations` to `__init__`, since it wont ever change during execution.